### ISSUE TITLE
Autocorrecting Lint/RedundantCopDisableDirective on method definition moves next line up

### ISSUE THREAD

**Issue by johansenja** [rubocop/rubocop#10114]
When you have a redudant dop disable on the same line as a `def` method definition, it strips the new line and interprets the next line as arguments

--------

## Expected behavior

```ruby
def foo # rubocop:todo Metrics/PerceivedComplexity
  bar
end
```

should get corrected to

```ruby
def foo
  bar
end
```

## Actual behavior

```ruby
def foo # rubocop:todo Metrics/PerceivedComplexity
  bar
end
```

gets corrected to

```ruby
def foo(bar)
end
```

#### And this also happens when the next line has args, which then results in a syntax error:

```ruby
def foo # rubocop:todo Metrics/PerceivedComplexity
  bar(1)
end
```

gets corrected to

```ruby
def foo  bar(1)
end
```

## Steps to reproduce the problem

`vi foo.rb`

```ruby
# include an unnecessary disable directive
def foo # rubocop:todo Metrics/PerceivedComplexity
  bar
end
```

`vi .rubocop.yml`

```yml
# ...
Metrics/PerceivedComplexity:
  Enabled: false
# ...
```

`rubocop --debug -a foo.rb`:

```
For [project dir]: configuration from [project dir]/.rubocop.yml
Default configuration from ~/.rvm/gems/ruby-2.7.1/gems/rubocop-1.7.0/config/default.yml
Inspecting 1 file
Scanning [project dir]/foo.rb
Loading cache from ~/.cache/rubocop_cache/f5ab2e2affe2e31a11b1e9fb9051c4ff8b782ff8/0536d18c9eb040c724be8df71fd74a67730b58ef/20d573ed671d38689e9ae4db0c2c7d11a8e4c0fb
W

Offenses:

foo.rb:1:9: W: [Corrected] Lint/RedundantCopDisableDirective: Unnecessary disabling of Metrics/PerceivedComplexity.
def foo # rubocop:todo Metrics/PerceivedComplexity
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
foo.rb:1:11: C: [Corrected] Style/MethodDefParentheses: Use def with parentheses when there are parameters.
def foo   bar
          ^^^

1 file inspected, 2 offenses detected, 2 offenses corrected
Finished in 0.1291950000450015 seconds
```

## RuboCop version

```
$ bundle exec rubocop -V
1.7.0 (using Parser 2.7.2.0, rubocop-ast 1.4.1, running on ruby 2.7.1 x86_64-darwin19)
```
I realise this is a few versions vehind, though bundler doesn't want to update rubocop any further than this (with `bundle update rubocop`) based on current config (for instance if I specify `gem 'rubocop', "1.21"` I get some `parser` version mismatch errors). Apologies if this has been addressed in later versions (though I did scan open and closed issues and it didn't appear this was already reported)

**Comment by tejasbubane** [comment]
@johansenja I could not reproduce this with latest version of rubocop:

```sh
$ cat test.rb
def foo # rubocop:todo Metrics/PerceivedComplexity
  bar(1)
end

$ cat .rubocop.yml 
Metrics/PerceivedComplexity:
  Enabled: false

$ rubocop -a test.rb
....
test.rb:1:9: W: [Corrected] Lint/RedundantCopDisableDirective: Unnecessary disabling of Metrics/PerceivedComplexity.
def foo # rubocop:todo Metrics/PerceivedComplexity
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

$ cat test.rb       
def foo
  bar(1)
end

$ rubocop -V        
1.21.0 (using Parser 3.0.2.0, rubocop-ast 1.11.0, running on ruby 2.7.2 x86_64-linux)
```
Please try to find a way to upgrade rubocop.

**Comment by koic** [comment]
This issue was resolved in the latest RuboCop 1.59.
